router-tests/nats_events_test.go

```diff
@@ -256,6 +256,121 @@ func TestNatsEvents(t *testing.T) {
 		})
 	})
 
+	t.Run("multipart", func(t *testing.T) {
+		assertLineEquals := func(reader *bufio.Reader, expected string) {
+			line, _, err := reader.ReadLine()
+			require.NoError(t, err)
+			require.Equal(t, expected, string(line))
+		}
+
+		assertMultipartPrefix := func(reader *bufio.Reader) {
+			assertLineEquals(reader, "--graphql")
+			assertLineEquals(reader, "Content-Type: application/json")
+			assertLineEquals(reader, "")
+		}
+
+		heartbeatInterval := 5 * time.Second
+
+		t.Run("subscribe with multipart responses", func(t *testing.T) {
+			t.Parallel()
+
+			testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+
+				subscribePayload := []byte(`{"query":"subscription { employeeUpdated(employeeID: 3) { id details { forename surname } } }"}`)
+
+				wg := &sync.WaitGroup{}
+				wg.Add(1)
+
+				var client http.Client
+				go func() {
+					client = http.Client{
+						Timeout: time.Second * 100,
+					}
+					req, err := http.NewRequest(http.MethodPost, xEnv.GraphQLMultipartResponsesURL(), bytes.NewReader(subscribePayload))
+					require.NoError(t, err)
+
+					req.Header.Set("Content-Type", "application/json")
```

Comment:
According to the spec, the response should include `Transfer-Encoding: chunked` header for HTTP2 this header is disallowed.

Ref: https://www.apollographql.com/docs/router/executing-operations/subscription-multipart-protocol/